www/pages/register.js

```diff
@@ -18,13 +21,56 @@ class Register extends Component {
     this.state = initialState;
 
     this.handleChange = this.handleChange.bind( this );
+    this.handleChangeUsername = this.handleChangeUsername.bind( this );
+    this.handleChangeEmail = this.handleChangeEmail.bind( this );
     this.handleSubmit = this.handleSubmit.bind( this );
   }
 
   handleChange( event ) {
     this.setState({
       [ event.target.name ]: event.target.value
     });
+
+  }
+
+  handleChangeUsername( event ) {
+    this.setState({
+      [ event.target.name ]: event.target.value
+    });
+
+    checkExistingUsername( event.target.value )
+    .then( result => {
+      if ( result === true ) {
+        this.setState({
+          showUsernameError: true
+        });
+      } else {
+        this.setState({
+          showUsernameError: false
+        });
+      }
+    });
+  }
+
+  handleChangeEmail( event ) {
+    this.setState({
+      [ event.target.name ]: event.target.value
+    });
+
+    checkExistingEmail( event.target.value )
+    .then( result => {
+      if ( result === true ) {
+        this.setState({
+          showEmailError: true
+        });
+      } else {
+        this.setState({
+          showEmailError: false
+        });
+      }
+
+      console.log( result );
+    });
   }
 
   handleSubmit( event ) {
```

Comment:
We should disable the submit button until the two checks have passed.

Reply:
To do that you would need to add three values to the state `loading`, `uniqueUser`, and `uniqueEmail`. While the checks are running you set `loading` to true. The submit button is disabled while `loading` is true or while either `uniqueUser` or `uniqueEmail is false.`